### DISCUSSION TITLE
MacOS global hotkey toggles all tabs in to separate windows

### DISCUSSION THREAD

**Discussion by rblalock** [ghostty-org/ghostty#3372] [Feature Requests, Ideas] [answered]
I attached a video to show the issue.  This happens on latest macos for me when I add this to the config:

keybind    = global:shift+super+t=toggle_visibility

After playing around with it - it seems to happen if i have had multiple tabs open at a given time (even if I had closed them).  jthey will all be recreated as separate windows when I toggle the visibility.  If I start Ghostty anew, no prior tabs, the toggle works fine but as soon as I add a tab, then toggle viz, I get these ghost windows like you see in the video.

https://github.com/user-attachments/assets/fa50bada-aad7-4665-a7c5-eddf3a2aef67

**Comment by rix1** [comment]
Can confirm - I'm able to reproduce. Although no problem if you haven't opened two or more tabs in the current session.

Here's my config:
```
keybind=global:unconsumed:option+z=toggle_visibility
```

![CleanShot 2024-12-28 at 00 08 25@2x](https://github.com/user-attachments/assets/3bdaef03-5ed1-4d70-9c21-f10ff5f81ede)

**Comment by mitchellh** [comment, marked as answer]
#4329
